dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/NativeHttpRestClient.java

```diff
@@ -0,0 +1,386 @@
+                        namedXContentEntries.stream())
+                        .flatMap(Function.identity()).collect(toList()));
+    }
+
+    /**
+     * Loads and returns the {@link NamedXContentRegistry.Entry} parsers provided by plugins.
+     */
+    static List<NamedXContentRegistry.Entry> getProvidedNamedXContents() {
+        List<NamedXContentRegistry.Entry> entries = new ArrayList<>();
+        for (NamedXContentProvider service : ServiceLoader.load(NamedXContentProvider.class)) {
+            entries.addAll(service.getNamedXContentParsers());
+        }
+        return entries;
+    }
+
+    public SearchResponse search(SearchRequest searchRequest, RequestOptions options) {
+
+        final HttpGetWithEntity httpGetWithEntity = new HttpGetWithEntity();
+
+        try {
+
+            httpGetWithEntity.setURI(new URI(url  + "/" + searchRequest.indices()[0] + "/_search"));
+
+            httpGetWithEntity.setEntity(createEntity(searchRequest.source(), REQUEST_BODY_CONTENT_TYPE));
+
+            final CloseableHttpResponse response = client.execute(httpGetWithEntity);
+
+            if (response.getStatusLine().getStatusCode() != HttpStatus.SC_OK) {
+                System.err.println("Method failed: " + response.getStatusLine());
+            }else{
+                return parseEntity(response.getEntity(), SearchResponse::fromXContent);
+            }
+
+        } catch (IOException e) {
+            System.err.println("Fatal transport error: " + e.getMessage());
+            e.printStackTrace();
+        } catch (URISyntaxException e) {
+            e.printStackTrace();
+        } finally {
+            // Release the connection.
+            httpGetWithEntity.releaseConnection();
+        }
+
+        return null;
+    }
+
+    HttpEntity createEntity(ToXContent toXContent, XContentType xContentType) throws IOException {
+        return createEntity(toXContent, xContentType, ToXContent.EMPTY_PARAMS);
+    }
+
+    HttpEntity createEntity(ToXContent toXContent, XContentType xContentType, ToXContent.Params toXContentParams)
+            throws IOException {
+        BytesRef source = XContentHelper
+                .toXContent(toXContent, xContentType, toXContentParams, false).toBytesRef();
+        return new NByteArrayEntity(source.bytes, source.offset, source.length, createContentType(xContentType));
+    }
+
+    /**
+     * Ignores deprecation warnings. This is appropriate because it is only
+     * used to parse responses from Elasticsearch. Any deprecation warnings
+     * emitted there just mean that you are talking to an old version of
+     * Elasticsearch. There isn't anything you can do about the deprecation.
+     */
+    private static final DeprecationHandler DEPRECATION_HANDLER = new DeprecationHandler() {
+        @Override
+        public void usedDeprecatedName(String usedName, String modernName) {}
+        @Override
+        public void usedDeprecatedField(String usedName, String replacedWith) {}
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'usedName' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5121287)